csa.rs

```diff
@@ -100,6 +98,6 @@ fn main() {
                    let arrival_station = params[1] as usize;
                    let departure_time = params[2];
 
-                   csa_compute(&timetable, departure_station, arrival_station, departure_time);
+                   csa_compute(&timetable[..], departure_station, arrival_station, departure_time);
```

Comment:
That's note needed, no? And using a collect for collecting a Vec<()> is code obfuscation and useless memory allocation. Better to use a for loop, it's not evil.